scripts/benchmark/utils/chain_plumbing.py

```diff
@@ -70,56 +67,33 @@
     'nonce': constants.GENESIS_NONCE
 }
 
+DEFAULT_GENESIS_STATE = [
+    (FUNDED_ADDRESS, {
+        "balance": DEFAULT_INITIAL_BALANCE,
+        "code": b'',
+    }),
+    (SECOND_ADDRESS, {
+        "balance": DEFAULT_INITIAL_BALANCE,
+        "code": b'',
+    }),
+]
 
-@to_dict
-def genesis_state(setup: Iterable[AddressSetup]) -> Any:
-    for value in setup:
-        yield value.address, {
-            "balance": value.balance,
-            "nonce": 0,
-            "code": value.code,
-            "storage": {}
-        }
+GenesisState = Iterable[Tuple[Address, Dict[str, Any]]]
 
 
-def chain_without_pow(
-        base_db: AtomicDB,
-        vm: Type[BaseVM],
-        genesis_params: Any,
-        genesis_state: Any) -> MiningChain:
+def get_chain(vm: Type[BaseVM], genesis_state: GenesisState) -> MiningChain:
 
-    vm_without_pow = vm.configure(validate_seal=lambda block: None)
+    chain = build(
+        MiningChain,
+        fork_at(vm, constants.GENESIS_BLOCK_NUMBER),
+        disable_pow_check(),
+        genesis(params=GENESIS_PARAMS, state=genesis_state)
+    )
 
-    klass = MiningChain.configure(
-        __name__='TestChain',
-        vm_configuration=(
-            (constants.GENESIS_BLOCK_NUMBER, vm_without_pow),
-        ))
-    chain = klass.from_genesis(base_db, genesis_params, genesis_state)
     return chain
 
 
-def get_chain(vm: Type[BaseVM]) -> MiningChain:
-    return chain_without_pow(
-        AtomicDB(),
-        vm,
-        GENESIS_PARAMS,
-        genesis_state([
-            AddressSetup(
-                address=FUNDED_ADDRESS,
-                balance=DEFAULT_INITIAL_BALANCE,
-                code=b''
-            ),
-            AddressSetup(
-                address=SECOND_ADDRESS,
-                balance=DEFAULT_INITIAL_BALANCE,
-                code=b''
-            ),
-        ])
-    )
-
-
-def get_all_chains() -> Iterable[MiningChain]:
+def get_all_chains(genesis_state: GenesisState=DEFAULT_GENESIS_STATE) -> Iterable[MiningChain]:
     for vm in ALL_VM:
-        chain = get_chain(vm)
+        chain = get_chain(vm, DEFAULT_GENESIS_STATE)
```

Comment:
I suppose this was intended to be `get_chain(vm, genesis_state)`.

Reply:
When did I become so sloppy... :see_no_evil: https://github.com/ethereum/py-evm/pull/1376